### PR TITLE
Prometheus: update to 0.14.0

### PR DESCRIPTION
**Pull request by benley**
Blog post: http://prometheus.io/blog/2015/06/01/advanced-service-discovery/

### CODE REVIEW

File: pkgs/top-level/go-packages.nix [hunk 1]
```diff
@@ -2416,13 +2429,26 @@ let
...
-      url = "https://github.com/go-yaml/yaml.git";
+      owner = "go-yaml";
+      repo = "yaml";
       sha256 = "0jbdy41pplf2d1j24qwr8gc5qsig6ai5ch8rwgvg72kq9q0901cy";
     };
   };
 
+  yaml-v2 = buildGoPackage rec {
+    rev = "c1cd2254a6dd314c9d73c338c12688c9325d85c6";
+    name = "yaml-v2-${stdenv.lib.strings.substring 0 7 rev}";
```
Comment (jagajaga) [inline pkgs/top-level/go-packages.nix]: https://github.com/NixOS/nixpkgs/blob/master/doc/coding-conventions.xml#L259
Once again :)

Comment (benley) [inline pkgs/top-level/go-packages.nix]: I was just matching what nearly every other package in that file does, on the assumption that internal consistency was more important than strictly following the style guide.  I'll happily change it to a date string if you prefer, though.

**Comment by benley** [comment]
I have no idea what's causing this build failure - I can't reproduce it!

**Comment by lethalman** [comment]
Built locally, thanks.

**Pull request merged**